### PR TITLE
feat: Export a decoupled Sass importer

### PR DESCRIPTION
**Pull request by vvanpo**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [x] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

See #873

### CODE REVIEW

File: src/importer.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  return (url, prev, done) => {
+    resolve(prev, url)
+      .then((result) => {
+        done({ file: result });
+      })
+      .catch(() => {
+        done(null);
+      });
+  };
+}
```
Comment (evilebottnawi) [inline src/importer.js:73]: I think it should be on your code side, not here, we already provide `getSassImplementation` and `getWebpackResolver` for you

File: src/utils.js [hunk 2]
```diff
@@ -262,75 +259,83 @@ export default function getPossibleRequests(
...
-const matchCss = /\.css$/i;
+function promiseResolve(callbackResolve) {
+  return (context, request) =>
+    new Promise((resolve, reject) => {
+      callbackResolve(context, request, (err, result) => {
+        if (err) reject(err);
+        else resolve(result);
+      });
+    });
+}
```
Comment (evilebottnawi) [inline src/utils.js:270]: I think we need to solve it on webpack side, but it is out of scope this PR, so let's do it in future

File: src/importer.js (hunk 1 above)
Comment (vvanpo) [reply src/importer.js:73]: OK, I'll remove this file. So I should just import directly from `dist/utils.js`, then? How should I document those functions to make sure that they aren't moved or their signature changed in future refactors?

File: src/utils.js (hunk 2 above)
Comment (vvanpo) [reply src/utils.js:270]: Yeah I had to do this to be compatible with `enhanced-resolve`. `loaderContext.getResolve()` [returns a promise-based function](https://github.com/webpack/webpack/blob/v4.44.1/lib/NormalModule.js#L205), but `enhanced-resolve` does not, unfortunately.

File: src/importer.js (hunk 1 above)
Comment (evilebottnawi) [reply src/importer.js]: Just put comments above functions with something /* WARNING: reason */

**Comment by vvanpo** [comment]
@evilebottnawi I had been ignoring the broken build because the only regression in code coverage was due to two parameters being made optional, and not calling them specifically without arguments gets reported as missing branch coverage. But I realize now that you were probably waiting on me to fix the build. So I've added some tests to test the `getWebpackResolver` function separately, not only to fix the coverage regression but also to give us some protection against that function's API being accidentally changed in future refactors.

File: src/utils.js [hunk 3]
```diff
@@ -262,23 +264,53 @@ export default function getPossibleRequests(
...
       return Promise.reject();
     }
 
     const [{ resolve, context, possibleRequests }] = resolutionMap;
 
-    let result;
-
     try {
-      result = await resolve(context, possibleRequests[0]);
+      return await resolve(context, possibleRequests[0]);
```
Comment (evilebottnawi) [inline src/utils.js:313]: No need `await` for `return`, just remove `await`

Comment (vvanpo) [reply src/utils.js:313]: If I don't `await` the promise, the `try ... catch` block will never catch any promise rejections, so a lot of tests fail.

**Review by evilebottnawi — APPROVED** [review]
Sorry for delay, hope we don't break something

File: src/utils.js (hunk 3 above)
Comment (evilebottnawi) [reply src/utils.js:313]: I will do refactor in future ()

**Comment by evilebottnawi** [comment]
I will not changelog it, because it is internal

**Pull request merged**